src/main/java/org/spongepowered/api/event/action/InteractEvent.java

```diff
@@ -24,14 +24,69 @@
  */
 package org.spongepowered.api.event.action;
 
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.data.type.HandType;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
 import org.spongepowered.api.event.Event;
-import org.spongepowered.math.vector.Vector3d;
+import org.spongepowered.api.event.EventContextKeys;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
 
 import java.util.Optional;
 
 /**
  * Base event for all interactions.
  */
 public interface InteractEvent extends Event {
-
+    /**
+     * Gets the fake {@link Player} who performed interaction.
+     *
+     * @return Fake player performing interaction
+     */
+    default Optional<Player> fakePlayer() {
```

Comment:
Is this really something that should be here? If so, this probably needs to either have (or point to) more documentation of what a fake player actually is, when one is available, etc.